File: app/src/main/java/com/larryhsiao/nyx/attachments/NewAttachments.java
package com.larryhsiao.nyx.attachments;

import com.silverhetch.clotho.Source;

import java.sql.Connection;
import java.util.ArrayList;
import java.util.List;

/**
 * Create multiple attachments
 */
public class NewAttachments implements Source<List<Attachment>> {
    private final Source<Connection> connSource;
    private final long jotId;
    private final String[] uris;

    public NewAttachments(Source<Connection> connSource, long jotId, String[] uris) {
        this.connSource = connSource;
        this.jotId = jotId;
        this.uris = uris;
    }

    @Override
    public List<Attachment> value() {
        List<Attachment> res = new ArrayList<>();
        for (String uri : uris) {
            res.add(new NewAttachment(
                connSource,
                uri,
                jotId
            ).value());
        }
        return res;
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/attachments/NewAttachments.kt
package com.larryhsiao.nyx.core.attachments

import com.silverhetch.clotho.Source
import java.sql.Connection
import java.util.*

/**
 * Create multiple attachments
 */
class NewAttachments(
    private val connSource: Source<Connection>,
    private val jotId: Long,
    private val uris: Array<String>
) : Source<List<Attachment?>?> {
    override fun value(): List<Attachment?> {
        val res: MutableList<Attachment?> = ArrayList()
        for (uri in uris) {
            res.add(NewAttachment(
                connSource,
                uri,
                jotId
            ).value())
        }
        return res
    }
}